ufl/algorithms/formfiles.py

```diff
@@ -204,7 +204,7 @@ def get_form(name):
     # Validate types
     if not isinstance(ufd.expressions, (list, tuple)):
         error("Expecting 'expressions' to be a list or tuple, not '%s'." % type(ufd.expressions))
-    if not all(isinstance(e, Expr) for e in ufd.expressions):
+    if not all(isinstance(e[0], Expr) for e in ufd.expressions):
```

Comment:
I guess I don't know how the form files work, but should one not write:

```
all(isinstance(e, Expr) for e in itertools.chain(*ufd.expressions))
```

Rather than implicitly indicating that the `ufd.expressions` variable is a list of singleton lists.

Reply:
Currently, "expressions" is a list of tuples of different type, `List[Tuple(ufl.core.expr.Expression, numpy.ndarray)]`, but this is not standardised within UFL. We only need to know that first item in the tuple is of UFL Expression type. The second item in the tuple is used to tell what reference points expression is evaluated.